configure-oidc.md

```diff
@@ -75,3 +75,128 @@ When issuing a `cosign sign` command, add the flags to pass in the client secret
+
+```shell
+$ oc get routes -n keycloak-system | awk '/keycloak/ {print $2}' | head -1
+keycloak-keycloak-system.apps.rosa.umebj-fwkt3-8j8.y6tr.p3.openshiftapps.com
+```
+Copy this route, adding `https://` in front, into your browser.
+
+You will need to log in. The username will be `admin`. You can retrieve the password with:
+```shell
+$ oc get secret/credential-keycloak -n keycloak-system -o yaml | awk '/ADMIN_PASSWORD:/ {print $2}' | base64 -d; printf "\n"
+```
+
+### Add Github Identity Provider to Keycloak Realm
+
+Select one of the options below to set up identity federation to the Github Identity Provider. 
+
+#### Option 1: Create a KeycloakRealm in OpenShift on the CLI
+
+Create a Custom Resource file from which to create the KeycloakRealm that includes the Github identity provider:
+
+```shell
+$ cat realm-cr.yaml
+apiVersion: keycloak.org/v1alpha1
+kind: KeycloakRealm
+metadata:
+  labels:
+    app: sso
+  name: sigstore
+spec:
+  instanceSelector:
+    matchLabels:
+      app: sso
+  realm:
+    displayName: Sigstore
+    enabled: true
+    id: sigstore
+    realm: sigstore
+    sslRequired: none
+    identityProviders:
+      - alias: github
+        displayName: Github
+        enabled: true
+        firstBrokerLoginFlowAlias: first broker login
+        internalId: github
+        providerId: github
+        trustEmail: true
+        config:
+            clientId: <my client id> # add your client ID
+            clientSecret: <my client secret> # add your client secret
+```
+
+Make sure that you have no existing KeycloakRealm with the `metadata.name` specified in your `realm-cr.yaml` file. IMPORTANT: make sure you are okay with deleting your current `sigstore` KeycloakRealm.
+```shell
+$ oc delete keycloakrealm sigstore -n keycloak-system
+keycloakrealm.keycloak.org "sigstore" deleted
+```
+
+Create the new KeycloakRealm that includes the Github identity provider from the custom resource file:
+```shell
+$ oc create -f realm-cr.yaml -n keycloak-system
+keycloakrealm.keycloak.org/sigstore created
+```
+
+#### Option 2: 
+You may also add the Github Identity provider to an existing KeycloakRealm through the Keycloak console by selecting `Identity Providers` in the lefthand menu bar and selecting `Add provider -> Github` from the drop down menu. See above for instructions on how to access the Keycloak console.
+
+### Copy the Redirect URI to your Github OAuth App
+
+From the Keycloak console, navigate to `Identity Providers` in the lefthand menu bar, and select the provider you have just created: `github`. It should look like this:
+
+<p align="center">
+  <img src="/images/github_idp.png" />
+</p>
+
+From this page, copy the Redirect URI:
+
+<p align="center">
+  <img src="/images/redirect_uri.png" />
+</p>
+
+Paste the redirect URI value into the OAuth app Authorization Callback URL field, then click `Update Application`:
+<p align="center">
+  <img src="/images/github_callback.png" />
+</p>
+
+### Add IdentityProviderMapper
```

Comment:
Can we configure this via the `KeycloakRealm` CR to automate the process?

Reply:
That's what I was trying to figure out. As far as I could tell, there was no way to do it with the RHSSO Operator. Adding an `IdentityProviderMapper` or `IdentityProviderMappers` key was either ignored or made the KeycloakRealm fail to come up. If you know of a way to do this, I would love to add it.